### PR TITLE
Remove ssl as a default feature

### LINKED ISSUE #409: Change SSL to opt-in
Currently the SSL library is opt-out, having to manually specify `default-features=false` in every crate that depends on Iron.

Many of the middleware available does not care about whether SSL is enabled or not. However, many libraries do not opt out of the feature, making compilation without it almost impossible. This is especially an issue on Windows where compiling openssl-sys is not easy.

#351 did not take a clear stand on inclusion of this feature. I would like to see this changed to the opt-in behaviour.

Sample middleware which are currently unsusable without ssl:
* [handlebars-iron](https://github.com/sunng87/handlebars-iron/blob/master/Cargo.toml)
* [router](https://github.com/iron/router) (present in TOML, but not yet pushed to crates.io)
* [mount](https://github.com/iron/mount/blob/master/Cargo.toml)
* [staticfile](https://github.com/iron/staticfile/blob/master/Cargo.toml)

### PR DESCRIPTION
**Pull request by mvdnes**
Resolves #409

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ rust:
     - nightly
 sudo: false
 script:
-    - cargo build
-    - cargo test
+    - cargo build --features=ssl
+    - cargo test --features=ssl
```
Comment (reem) [inline .travis.yml]: Can you add `cargo build` and `cargo test` without `ssl` as well?

**Comment by mvdnes** [comment]
Pushed a commit for testing with and without the ssl feature.

**Comment by untitaker** [comment]
@reem? LGTM

**Comment by mvdnes** [comment]
Great, thanks! :+1:

**Pull request merged**